### PR TITLE
*: remove Avro OCF sources and sinks

### PR DESCRIPTION
**Pull request by benesch**
Avro OCF sources and sinks don't work in Materialize Cloud, so remove
them. See #11875 for details.

Co-authored-by: Gus Wynn <gus@materialize.com>

### Motivation

* This PR removes code.

### Testing

- [x] This PR has adequate test coverage / QA involvement has been duly considered.

### Release notes

This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - n/a

### CODE REVIEW

File: test/coordtest/cancel.ct [hunk 1]
```diff
@@ -1,17 +0,0 @@
...
-async-sql session=now_1h
-SELECT * FROM t AS OF now()+'1h';
-----
-
-async-cancel session=now_1h
-----
-
-await-sql session=now_1h
-----
-Canceled
```
Comment (guswynn) [inline test/coordtest/cancel.ct:17]: this one probably shouldn't have been deleted?

File: test/coordtest/github-6687.ct [hunk 2]
```diff
@@ -1,48 +0,0 @@
...
-
-sql
-CREATE MATERIALIZED VIEW v AS SELECT * FROM t;
-----
-CreatedView {
-    existed: false,
-}
-
-sql
-CREATE SINK s FROM v INTO AVRO OCF '<TEMP>/sink';
```
Comment (guswynn) [inline test/coordtest/github-6687.ct:44]: make this a kafka sink instead?

File: test/coordtest/linearizability.ct [hunk 3]
```diff
@@ -1,107 +0,0 @@
-inc-timestamp
```
Comment (guswynn) [inline test/coordtest/linearizability.ct:1]: why was this file deleted?

**Review by guswynn — APPROVED** [review]
a few comments about some deleted tests, but looks good!

also, my original commit I believed left the acro ocf decode/encode code unused, and I think I intended on adding a comment saying why we still have that dead code (for future extensions, like ocf-on-s3). might be worth adding a comment like that in this pr!

**Comment by benesch** [comment]
Those coordtest files are cruft from #11329 that I missed when I removed it!

> also, my original commit I believed left the acro ocf decode/encode code unused, and I think I intended on adding a comment saying why we still have that dead code (for future extensions, like ocf-on-s3). might be worth adding a comment like that in this pr!

Ah, as a policy we don't leave dead code around! We've got the Git log if we need to resurrect it eventually. But dead code bogs down refactorings, and it's always a bit wasteful to refactor dead code, since you have no way to test whether it works.
